server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -450,17 +549,42 @@ public Consumer getHypervisor(String hypervisorId, Owner owner) {
     }
 
     /**
+     * Lookup all registered consumers matching one of the given hypervisor IDs.
+     *
+     * Results are returned as a map of hypervisor ID to the consumer record created.
+     * If a hypervisor ID is not in the map, this indicates the hypervisor consumer does
+     * not exist, i.e. it is new and needs to be created.
+     *
+     * This is an unsecured query, manually limited to an owner by the parameter given.
+     * @param owner Owner to limit results to.
+     * @param hypervisorIds List of hypervisor IDs as reported by the virt fabric.
+     *
+     * @return VirtConsumerMap of hypervisor ID to it's consumer, or null if none exists.
+     */
+    @Transactional
+    public VirtConsumerMap getHostConsumersMap(Owner owner,
+            Collection<String> hypervisorIds) {
+        List<Consumer> results = getHypervisorsBulk(hypervisorIds, owner.getKey());
+        VirtConsumerMap hypervisorMap = new VirtConsumerMap();
+        for (Consumer c : results) {
+            hypervisorMap.add(c.getHypervisorId().getHypervisorId(), c);
+        }
+        return hypervisorMap;
+    }
+
+    /**
      * @param hypervisorIds list of unique hypervisor identifiers
      * @param ownerKey Org namespace to search
      * @return Consumer that matches the given
      */
     @SuppressWarnings("unchecked")
     @Transactional
-    public List<Consumer> getHypervisorsBulk(List<String> hypervisorIds, String ownerKey) {
+    public List<Consumer> getHypervisorsBulk(Collection<String> hypervisorIds,
+            String ownerKey) {
         if (hypervisorIds == null || hypervisorIds.isEmpty()) {
             return new LinkedList<Consumer>();
         }
-        return createSecureCriteria()
+        return currentSession().createCriteria(Consumer.class)
```

Comment:
Because a hypervisor checkin is done as a consumer, they typically cannot see other consumers, so with secure criteria the automatic filtering kicks in and hides all the host consumers. This is why the singular getHost does unsecured as well. This actually tripped me up for hours trying to figure out why it wasn't seeing the hosts and trying to re-create them. :)